### ISSUE TITLE
VoiceOver on MacOS stops reading the code editor after jumping to the end of a long code file

### ISSUE THREAD

**Issue by Ghorthalon** [microsoft/vscode#294263]
Type: <b>Bug</b>

When using VoiceOver on the Mac, I have tested 26.2 stable as well as 26.3 RC, when jumping to the end of a code editor with cmd+down, VoiceOver says something such as "Selection replaced right brace". From then on, VoiceOver will not read the code editor anymore until you open a dialog like the command pallet and escape back out. Switching the focus to the project explorer and back by pressing cmd shift e twice does not work, same goes for the outline explorer, etc. The most reliable way to get it to read again I have found is to press cmd shift p and escape. 

To reproduce:
1. Load a source file of about 100 or more lines.
2. Either arrow down through the entire file, or press cmd+down arrow to jump to the end.
3. Observe that from then on, VoiceOver will not read the code editor anymore. You can arrow up and down, but VoiceOver remains silent.
4. Press cmd shift p to open the command pallet, then escape to focus back to the editor. Now, the editor will read again, until you reach the end once more, at which point it will stop reading again.

It is possible that there are other instances in which it stops speaking, but this is by far the most common one I have found yet. 

VS Code version: Code 1.109.2 (Universal) (591199df409fbf59b4b52d5ad4ee0470152a9b31, 2026-02-10T20:18:23.520Z)
OS version: Darwin arm64 25.3.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M2 Max (12 x 2400)|
|GPU Status|2d_canvas: enabled<br>GPU0: VENDOR= 0x106b [Google Inc. (Apple)], DEVICE=0x0000 [ANGLE (Apple, ANGLE Metal Renderer: Apple M2 Max, Version 26.3 (Build 25D122))], DRIVER_VENDOR=Apple, DRIVER_VERSION=26.3 *ACTIVE*<br>Machine model name: Mac<br>Machine model version: 14.5<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 3, 3|
|Memory (System)|32.00GB (0.13GB free)|
|Process Argv|--crash-reporter-id 17c62c91-76c5-4b04-b5f0-c871bbd53fba|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (73)</summary>

Extension|Author (truncated)|Version
---|---|---
add-reference|adr|1.0.2
asp-net-core-switcher|adr|2.0.2
blazor-snippet-pack|adr|2.6.1
csharp-to-typescript|adr|1.12.1
libman|adr|1.0.1
namespace|adr|1.1.2
terminal-commands|adr|1.0.5
toggle-hidden|adr|1.0.2
user-secrets|adr|2.0.1
odin|aes|1.2.2
nelua|ale|0.1.0
path-intellisense|chr|2.10.0
gitignore|cod|0.10.0
dart-code|Dar|3.128.0
flutter|Dar|3.128.0
vscode-deno|den|3.50.0
netcore-editorconfiggenerator|dog|0.2.0
netcore-extension-pack|dog|1.10.0
netcore-snippets|dog|6.0.3
quicktype-refresh|dog|1.0.2
EditorConfig|Edi|0.18.1
dotnet|for|0.0.4
dotnet-test-explorer|for|0.7.8
copilot-chat|Git|0.37.4
go|gol|0.52.2
rest-client|hum|0.25.1
vscode-nuget-package-manager|jmr|1.1.6
cmake-language-support-vscode|jos|0.0.9
docomment|k--|1.0.1
csharpextensions|kre|1.7.3
lldb-dap|llv|0.4.1
csdevkit|ms-|2.10.3
csharp|ms-|2.120.3
dotnet-maui|ms-|1.13.20
vscode-dotnet-runtime|ms-|3.0.0
vscodeintellicode-csharp|ms-|2.2.3
debugpy|ms-|2025.18.0
python|ms-|2026.0.0
vscode-pylance|ms-|2025.10.4
vscode-python-envs|ms-|1.16.0
cmake-tools|ms-|1.22.27
cpptools|ms-|1.30.5
cpptools-extension-pack|ms-|1.3.1
vsliveshare|ms-|1.0.5959
indent-report|ogo|1.0.0
vscode-nuget-gallery|pat|1.2.4
vscode-versionlens|pfl|1.22.4
java|red|1.52.0
vscode-xml|red|0.29.0
luacheck|rog|1.0.0
rust-analyzer|rus|0.3.2786
ruby-extensions-pack|Sho|0.1.13
ruby-lsp|Sho|0.10.0
sorbet-vscode-extension|sor|0.3.46
swift-lang|ssw|1.12.0
lua|sum|3.17.1
svelte-vscode|sve|109.13.0
swift-vscode|swi|2.16.1
msbuild-project-tools|tin|0.6.7
vscode-lldb|vad|1.11.5
hammerspoon|vir|0.5.2
intellicode-api-usage-examples|Vis|0.2.9
vscodeintellicode|Vis|1.3.2
extension|vla|0.1.3
vscode-gradle|vsc|3.17.2
vscode-java-debug|vsc|0.58.5
vscode-java-dependency|vsc|0.27.0
vscode-java-pack|vsc|0.30.5
vscode-java-test|vsc|0.44.0
vscode-maven|vsc|0.45.1
vscode-todo-highlight|way|1.0.5
vscode-zig|zig|0.6.17
vscode-proto3|zxh|0.5.5

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythonrdcb7:31342333
6518g693:31436602
aj953862:31281341
6abeh943:31336334
cloudbuttont:31379625
3efgi100_wstrepl:31403338
use-responses-api:31390855
je187915:31401257
ec5jj548:31422691
cmp-ext-treat:31426748
cp_cls_t_966_ss:31454198
ge8j1254_inline_auto_hint_haiku:31431912
nes-autoexp-off:31446584
a5gib710:31434435
2984c849_gpt52_codex:31457542
rename_enabled:31436409
7a04d226_do_not_restore_last_panel_session:31438103
anthropic_thinking_c:31457102
31c7i317_no_change:31458090
h0hdh950:31428394
preserve_tokens:31444547
cp_cls_t_1081:31454832
copilot-nes-oct-trt:31432596
ia-use-proxy-models-svc:31452481
nes-slash-models-on:31441832
nes-set-to-on:31441941
a43f0574:31442827
18jd0202:31460037
c0ci8992_validated:31449878
edit_mode_visible:31459180
0a33d263:31458759
showingstats:31457202
reg-2ge67640:31455970
nes-extended-on:31455476
chat:31457767
ah:31460422

```

</details>

**Comment by adincebic** [comment]
I can confirm that this is still an issue on:

Version: 1.119.1
Commit: 974500e64f0d1cfdf7c9821a2a51c2cb3bf0e561
Date: 2026-05-12T01:20:22+09:00
Electron: 39.8.8
ElectronBuildId: 13870025
Chromium: 142.0.7444.265
Node.js: 22.22.1
V8: 14.2.231.22-electron.0
OS: Darwin arm64 25.5.0

I am pretty sure that this is caused by some change on macOS since it works fine on Seqoya. However can't say that it is a OS bug or not since there might be a change in some web accessibility API that chromium or VSCode needs to adapt to. Regardless it results in a bug with VSCode.

**Comment by adincebic** [comment]
@meganrogge @aiday-mar do you have any update here? It is difficult to work in VSCode after the introduction of this bug?
